torch/_decomp/decompositions.py

```diff
@@ -2528,8 +2532,9 @@ def _upsample_nearest2d_common(input, h_indices, w_indices):
     return result
 
 
-@register_decomposition(aten.upsample_nearest2d.default)
+@register_decomposition([aten.upsample_nearest2d.default, aten.upsample_nearest2d.out])
 @aten.upsample_nearest2d.default.py_impl(DispatchKey.Autograd)
+@out_wrapper()
```

Comment:
Test failures are real, it's expecting channels last input to give a channels last output when passing in a zero-size out tensor.
https://github.com/pytorch/pytorch/blob/310f6ab11a4b36fddab0bb309566d52834bff0f5/test/test_torch.py#L8048

But in `out_wrapper` it just resizes without using any specific memory format, 
https://github.com/pytorch/pytorch/blob/310f6ab11a4b36fddab0bb309566d52834bff0f5/torch/_prims_common/wrappers.py#L175

I'm guessing this is a problem for other decompositions as well, but this is the only function that tests for it.

Reply:
Looks like for `TensorIterator` the behavior is to give the resized tensor the same strides as the functional output, but I expect this isn't true for all operators and some likely just call `resize_`.

Reply:
We should then add a optional parameter to `out_wrapper` that implements this logic